### ISSUE TITLE
pathlib.PureWindowsPath.match() mishandles path anchors

### ISSUE THREAD

**Issue by barneygale** [python/cpython#101360]
`pathlib.PurePath.match()` doesn't use `fnmatch()` to match path anchors, contrary to the documentation. Instead, it considers the drive and root of the pattern separately; if either (or both) is specified in the pattern, it must *exactly* match the corresponding drive/root in the path.

This results in the following:

```python
>>> from pathlib import PureWindowsPath as P
>>> P('c:/b.py').match('*:/*.py')
False
>>> P('c:/b.py').match('c:*.py')
True
>>> P('c:/b.py').match('/*.py')
True
>>> P('//some/share/a.py').match('//*/*/*.py')
False
>>> P('//some/share/a.py').match('/*.py')
True
```

All of these results are wrong IMO.

### Linked PRs
* gh-101363

**Comment by barneygale** [comment]
I think I know why this behaviour is here: pathlib originally wouldn't parse patterns like `*:/` because it required drive letters to be alphabetic. That requirement was removed in 2ba0fd5767577954f331ecbd53596cd8035d7186.

**Comment by barneygale** [comment]
Fixed in #101363 / d401b20630965c0e1d2a5a0d60d5fc51aa5a8d80
